characterization/fits.py

```diff
@@ -0,0 +1,36 @@
+from __future__ import print_function
+import os
+import shutil
+import subprocess
+import sys
+import tempfile
+import thread
+
+
+def writer(fn, input):
+    try:
+        subprocess.check_call(["ng", "edu.harvard.hul.ois.fits.Fits", "-i", input, "-o", fn],
+                              stderr=subprocess.PIPE)
+    except subprocess.CalledProcessError as e:
+        sys.exit("FITS failed to execute and exited with status {}".format(e.returncode))
+
+
+def main(input):
+    tempdir = tempfile.mkdtemp()
+    try:
+        fn = os.path.join(tempdir, "fits")
+        os.mkfifo(fn)
+    except OSError as e:
+        sys.exit("Unable to create a FIFO at location {}: {}".format(fn, e))
+    else:
+        thread.start_new_thread(writer, (fn, input))
+        print(open(fn).read())
```

Comment:
The pipe will always be deleted, and the fd will be released when the script exits, so I don't think wrapping it in a `with` is too critical. I can change it though.